assignments/Untitled-3.js

```diff
@@ -0,0 +1,303 @@
+/*
+Object oriented design is commonly used in video games.  For this part of the assignment you will be implementing several constructor functions with their correct inheritance hierarchy.
+
+In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+
+At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
+
+Each constructor function has unique properties and methods that are defined in their block comments below:
+*/
+
+/*
+=== GameObject ===
+* createdAt
+* name
+* dimensions (These represent the character's size in the video game)
+* destroy() // prototype method that returns: `${this.name} was removed from the game.`
+*/
+class GameObject {
+  constructor(gameObjectAttributes) {
+    this.createdAt = gameObjectAttributes.createdAt;
+    this.name = gameObjectAttributes.name;
+    this.dimensions = gameObjectAttributes.dimensions;
+  }
+  destroy() {
+    return `${this.name} was removed from the game`;
+  }
+}
+
+/*
+function GameObject(gameObjectAttributes) {
+  this.createdAt = gameObjectAttributes.createdAt;
+  this.name = gameObjectAttributes.name;
+  this.dimensions = gameObjectAttributes.dimensions;
+}
+
+//Method from the GameObject  function constructor
+GameObject.prototype.destroy = function () {
+  return `${this.name} was removed from the game.`;
+}
+*/
+
+/*
+=== CharacterStats ===
+* healthPoints
+* takeDamage() // prototype method -> returns the string '<object name> took damage.'
+* should inherit destroy() from GameObject's prototype
+*/
+class CharacterStats {
+  constructor(characterStatsAttributes) {
+    this.healthPoints = characterStatsAttributes.healthPoints;
+    this.createdAt = characterStatsAttributes.createdAt;
+    this.name = characterStatsAttributes.name;
+    this.dimensions = characterStatsAttributes.dimensions;
```

Comment:
You do not need to add the `createdAt`, `name` and `dimensions` attributes in class `CharacterStats` since they're inherited from the class `GameObject` with the use of `super()` 😊